modules/features/discover/src/main/java/au/com/shiftyjelly/pocketcasts/discover/view/CollectionListRowAdapter.kt

```diff
@@ -62,31 +113,68 @@ internal class CollectionListRowAdapter(
+        return CollectionListViewHolder(
+            binding,
+            onItemClicked = { pageIndex, podcastIndex ->
+                val podcastSublist = getItem(pageIndex)
+                val podcast = podcastSublist.getOrNull(podcastIndex) as? DiscoverPodcast
+
+                if (podcast == null) return@CollectionListViewHolder
+
+                fromListId?.let {
+                    analyticsTracker.track(
+                        AnalyticsEvent.DISCOVER_LIST_PODCAST_TAPPED,
+                        mapOf(
+                            LIST_ID_KEY to it,
+                            PODCAST_UUID_KEY to podcast.uuid,
+                        ),
+                    )
+                }
+                onPodcastClicked(podcast, fromListId)
+            },
+            onHeaderClicked = {
+                onHeaderClicked()
+            },
+            onPodcastSubscribe = { pageIndex, podcastIndex ->
+                val podcastSublist = getItem(pageIndex)
+                val podcast = podcastSublist.getOrNull(podcastIndex) as? DiscoverPodcast
+                podcast?.let { onPodcastSubscribe(it, fromListId) }
+            },
+        )
+    }
 
-            if (podcast == null) return@CollectionListViewHolder
+    override fun onBindViewHolder(holder: CollectionListViewHolder, position: Int) {
+        val podcastSublist = getItem(position)
 
-            fromListId?.let {
-                analyticsTracker.track(
-                    AnalyticsEvent.DISCOVER_LIST_PODCAST_TAPPED,
-                    mapOf(
-                        LIST_ID_KEY to it,
-                        PODCAST_UUID_KEY to podcast.uuid,
-                    ),
-                )
+        holder.bind(podcastSublist, isSubscribeButtonVisible = position != 0)
+
+        if (position == 0) {
+            this.header?.let {
+                holder.binding.header.root.isVisible = true
+                holder.binding.header.lblTitle.text = it.title
+                holder.binding.header.lblSubtitle.text = it.subtitle
+                holder.binding.header.imageHeader.load(it.imageUrl)
+                holder.binding.header.root.contentDescription =
+                    it.title + " " + holder.binding.root.context.getString(LR.string.discover_collection_header_content_description_suffix)
+
+                val height = getPodcastsHeight(holder.binding.podcasts)
+                val layoutParams = holder.binding.root.layoutParams
+                layoutParams.height = height
+                holder.binding.root.layoutParams = layoutParams
             }
-            onPodcastClicked(podcast, fromListId)
+        } else {
+            holder.binding.header.root.isVisible = false
         }
     }
 
-    override fun onBindViewHolder(holder: CollectionListViewHolder, position: Int) {
-        val podcastSublist = getItem(position)
-        holder.bind(podcastSublist)
+    fun getPodcastsHeight(view: View): Int {
+        view.measure(0, 0)
```

Comment:
The comment above was great in explaining the `modifiedList` logic. Could we do the same here, as I wasn't sure what this line did?

Reply:
I missed this in my own review. I've added a comment for it